### ISSUE TITLE
`method_missing': undefined method `merge' 

### ISSUE THREAD

**Issue by okainov** [NARKOZ/gitlab#574]
Starting today, we've started to get failures while running Danger in our CI, error seems to be coming from `gitlab` gem. I see new release was out yesterday, so probably it has broken it, could you please take a look?
```

 $ danger --fail-on-errors=true --fail-if-no-pr=true
 /usr/local/bundle/gems/gitlab-4.16.0/lib/gitlab/objectified_hash.rb:37:in `method_missing': undefined method `merge' for #<Gitlab::ObjectifiedHash:0x000056044d9a94d8> (NoMethodError)
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/request_sources/gitlab.rb:87:in `block (2 levels) in mr_comments'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/request_sources/gitlab.rb:87:in `map'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/request_sources/gitlab.rb:87:in `block in mr_comments'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/request_sources/gitlab.rb:87:in `each'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/request_sources/gitlab.rb:87:in `flat_map'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/request_sources/gitlab.rb:87:in `mr_comments'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/request_sources/gitlab.rb:180:in `update_pull_request_with_inline_comments!'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/request_sources/gitlab.rb:173:in `update_pull_request!'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/danger_core/dangerfile.rb:260:in `post_results'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/danger_core/dangerfile.rb:288:in `run'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/danger_core/executor.rb:29:in `run'
 	from /usr/local/bundle/gems/danger-8.0.2/lib/danger/commands/runner.rb:73:in `run'
 	from /usr/local/bundle/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
 	from /usr/local/bundle/gems/danger-8.0.2/bin/danger:5:in `<top (required)>'
 	from /usr/local/bundle/bin/danger:23:in `load'
 	from /usr/local/bundle/bin/danger:23:in `<main>'
```

Docker image `ruby:2.7.1`

gem install:
```
 $ gem install danger-gitlab
 Successfully installed claide-1.0.3
 Successfully installed open4-1.3.4
 Successfully installed colored2-3.1.2
 Successfully installed cork-0.3.0
 Successfully installed nap-1.1.0
 Successfully installed claide-plugins-0.9.2
 Successfully installed rchardet-1.8.0
 Successfully installed git-1.7.0
 Successfully installed multipart-post-2.1.1
 Successfully installed faraday-1.0.1
 Successfully installed faraday-http-cache-2.2.0
 Successfully installed kramdown-2.3.0
 Successfully installed kramdown-parser-gfm-1.1.0
 Successfully installed public_suffix-4.0.5
 Successfully installed addressable-2.7.0
 Successfully installed sawyer-0.8.2
 Successfully installed octokit-4.18.0
 Successfully installed unicode-display_width-1.7.0
 Successfully installed terminal-table-1.8.0
 Successfully installed no_proxy_fix-0.1.2
 Successfully installed danger-8.0.2
 Successfully installed multi_xml-0.6.0
 Successfully installed mime-types-data-3.2020.0512
 Successfully installed mime-types-3.3.1
 When you HTTParty, you must party hard!
 Successfully installed httparty-0.18.1
 Successfully installed gitlab-4.16.0
 Successfully installed danger-git
```

**Comment by okainov** [comment]
Double-checked using fixed `4.15.0` version - no failure.

**Comment by fvelcker** [comment]
Same here, all our pipelines are failing.
I compared the tags (https://github.com/NARKOZ/gitlab/compare/v4.15.0...v4.16.0) and noticed I know nothing... So better off just leaving a comment...

**Comment by leolorenzoluis** [comment]
@okainov How did you specify to install 4.15.0 in the dependency graph when installing with `gem install danger-gitlab`?

**Comment by okainov** [comment]
>@okainov How did you specify to install 4.15.0 in the dependency graph when installing with gem install danger-gitlab?

Here you go

    gem install gitlab -v 4.15.0 && gem install danger-gitlab

**Comment by leolorenzoluis** [comment]
@okainov Thank you. I tried that, but it will still install 4.16.0 thus causing an error. This was my solution

```
gem uninstall gitlab -v 4.16.0 
```

**Comment by okainov** [comment]
@leolorenzoluis that's odd, as it works for us :)
```
$ gem install gitlab -v 4.15.0 && gem install danger-gitlab
 Successfully installed multi_xml-0.6.0
 Successfully installed mime-types-data-3.2020.0512
 Successfully installed mime-types-3.3.1
 When you HTTParty, you must party hard!
 Successfully installed httparty-0.18.1
 Successfully installed unicode-display_width-1.7.0
 Successfully installed terminal-table-1.8.0
 Successfully installed gitlab-4.15.0
 7 gems installed
 Successfully installed claide-1.0.3
 Successfully installed open4-1.3.4
 Successfully installed colored2-3.1.2
 Successfully installed cork-0.3.0
 Successfully installed nap-1.1.0
 Successfully installed claide-plugins-0.9.2
 Successfully installed rchardet-1.8.0
 Successfully installed git-1.7.0
 Successfully installed multipart-post-2.1.1
 Successfully installed faraday-1.0.1
 Successfully installed faraday-http-cache-2.2.0
 Successfully installed kramdown-2.3.0
 Successfully installed kramdown-parser-gfm-1.1.0
 Successfully installed public_suffix-4.0.5
 Successfully installed addressable-2.7.0
 Successfully installed sawyer-0.8.2
 Successfully installed octokit-4.18.0
 Successfully installed no_proxy_fix-0.1.2
 Successfully installed danger-8.0.2
 Successfully installed danger-gitlab-8.0.0
 20 gems installed
```

**Comment by brettha** [comment]
Same issue here... Installed in gem file overriding v 4.16.0

gem 'gitlab', '4.15.0'

**Comment by philipqnguyen** [comment]
Same issue here in 4.16.0
```
       undefined method `merge' for #<Gitlab::ObjectifiedHash:0x0000000004b399e0>
     # ./bundle/ruby/2.5.0/gems/gitlab-4.16.0/lib/gitlab/objectified_hash.rb:37:in `method_missing'
```

**Comment by rymai** [comment]
@NARKOZ I'm surprised `Gitlab::ObjectifiedHash` doesn't respond to `#merge` since it's a pretty valid use case, and I'm even more surprised that this is only discovered now. I've opened a MR to add support for `Gitlab::ObjectifiedHash#merge` if that's ok: https://github.com/NARKOZ/gitlab/pull/576

**Comment by smcgivern** [comment]
I think this happened for Danger specifically because it calls `#merge` on items inside an array: https://github.com/danger/danger/blob/v8.0.2/lib/danger/request_sources/gitlab.rb#L87

And https://github.com/NARKOZ/gitlab/pull/570 just made hashes inside an array an `ObjectifiedHash`, when before they weren't (which was a bug).

**Comment by NARKOZ** [comment]
Sorry for the breaking change introduced in 4.16.0.

This should be fixed in version 4.16.1 via 242cf2d

**Comment by rymai** [comment]
Thanks!
